### PR TITLE
x11-misc/slim : Upstream project forked, updated ebuilds

### PR DESCRIPTION
**Pull request by RobPearce**
This pull request updates the Simple Login Manager to the latest release from the slim-fork project, which has resurrected what was a dead project and fixed lots of bugs.
Closes: https://bugs.gentoo.org/832562
Closes: https://bugs.gentoo.org/727544
Closes: https://bugs.gentoo.org/832303
Closes: https://bugs.gentoo.org/580458
Closes: https://bugs.gentoo.org/803476
Closes: https://bugs.gentoo.org/732430
Closes: https://bugs.gentoo.org/756181

### CODE REVIEW

**Comment by RobPearce** [comment]
Would it help if I add myself as a proxy maintainer? I note that the Guide says to do so before committing the fixes but if I volunteer now, would that be OK?

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (ceamac) [inline x11-misc/slim/slim-1.3.9-r1.ebuild:93]: You can use the `readme.gentoo-r1` eclass to display a message only on first installation, to skip this `none` check logic altogether.

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (ceamac) [inline x11-misc/slim/slim-1.3.9-r1.ebuild:23]: The dependencies should be sorted alphabetically. So please move media-libs/ deps upwards.

File: x11-misc/slim/slim-1.4.0.ebuild
Comment (ceamac) [inline x11-misc/slim/slim-1.4.0.ebuild:8]: There is no `slim-9999.ebuild`, did you add one and forgot to push it?

**Review by ceamac — CHANGES_REQUESTED** [review]
Thank you for your work on this, I see you closed a lot of bugs.

I didn't install and use this new slim yet, so I may come with further feedback after I do.  It builds succesfully.  I noticed 1.4.0 is missing a few files that still exist in 1.3.9 -
`usr/share/slim/themes/original/{background.jpg,panel.png,slim.theme`.  Is this intentional?  Maybe upstream removed them?

Also - please squash your commits and then force push the changes.  Ideally we should have 1 commit for 1.3.9 and one for 1.4.0, but having only one for both is also acceptable.

Thank you!

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (ceamac) [inline x11-misc/slim/slim-1.3.9-r1.ebuild:23]:
```suggestion
	media-libs/libjpeg-turbo:=
```

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (ceamac) [inline x11-misc/slim/slim-1.3.9-r1.ebuild:18]: + `x11-libs/libXext`

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (ceamac) [reply x11-misc/slim/slim-1.3.9-r1.ebuild:93]: (this is optional - we can leave it for another time)

**Comment by ceamac** [comment]
I tried them both, they work.  I like version 1.4 better, it has a nice green theme.

File: x11-misc/slim/slim-1.3.6-r5.ebuild
Comment (RobPearce) [reply x11-misc/slim/slim-1.3.6-r5.ebuild:23]: OK, fair point. I got carried away fixing pkgdev warnings

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (RobPearce) [reply x11-misc/slim/slim-1.3.9-r1.ebuild:74]: Is it worth retaining (and, I note, fixing) the elog for a first install, or do you think that's superfluous?

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (RobPearce) [reply x11-misc/slim/slim-1.3.9-r1.ebuild:1]: I had a 1.3.9 locally and posted it on the bugzilla (#832562) before spotting a minor issue that I felt was worth patching. As that ebuild had been made public and offered for inclusion in the tree, I wanted to avoid confusion.

File: x11-misc/slim/slim-1.4.0.ebuild
Comment (RobPearce) [reply x11-misc/slim/slim-1.4.0.ebuild:8]: I use one locally for testing - I'm not sure whether it would be desirable to add it. Which way would you prefer I fix this?

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (RobPearce) [reply x11-misc/slim/slim-1.3.9-r1.ebuild:18]: I think that's only required for the "slimlock" tool, which is only built if PAM is enabled, so should this be `pam? (x11-lib/libXext)` ?

**Comment by RobPearce** [comment]
> I noticed 1.4.0 is missing a few files that still exist in 1.3.9 -
> `usr/share/slim/themes/original/{background.jpg,panel.png,slim.theme`.
Is it not the other way round? Upstream has a new default theme for 1.4.0 and those files are the old theme from 1.3.x retained for people who prefer pink to green.

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (ceamac) [reply x11-misc/slim/slim-1.3.9-r1.ebuild:18]: Sure.  If it's used only with pam, add it conditionally.

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (ceamac) [reply x11-misc/slim/slim-1.3.9-r1.ebuild:74]: There is also a pretty comprehensive (and slightly outdated) wiki page for SLIM: https://wiki.gentoo.org/wiki/SLiM so you can display a link instead or add it to that text.

File: x11-misc/slim/slim-1.4.0.ebuild
Comment (ceamac) [reply x11-misc/slim/slim-1.4.0.ebuild:8]: Add it and keep it in sync with the last version, no point in keeping it private since you already use it :)

**Comment by ceamac** [comment]
> > I noticed 1.4.0 is missing a few files that still exist in 1.3.9 -
> > `usr/share/slim/themes/original/{background.jpg,panel.png,slim.theme`.
> 
> Is it not the other way round? Upstream has a new default theme for 1.4.0 and those files are the old theme from 1.3.x retained for people who prefer pink to green.

Right, I diffed them in reverse and got confused.  Thanks for clarifying.

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (a17r) [reply x11-misc/slim/slim-1.3.9-r1.ebuild:1]: bugzilla attachments are irrelevant for portage ebuild revisions.

File: x11-misc/slim/slim-1.3.9-r1.ebuild
Comment (a17r) [reply x11-misc/slim/slim-1.3.9-r1.ebuild:23]: is it necessary though? (I realise libjpeg-turbo has a subslot, but not every revdep links against specific soversion)

Comment (ceamac) [reply x11-misc/slim/slim-1.3.9-r1.ebuild:23]: This is linked against `libjpeg.so.62`.  So I guess if the 62 changes it will need rebuilding.

**Comment by RobPearce** [comment]
Hopefully I've addressed all the issues and managed to squash the commits as requested.

File: x11-misc/slim/slim-1.3.9.ebuild
Comment (ceamac) [reply x11-misc/slim/slim-1.3.9.ebuild:22]: The rest of the dependencies need to be sorted too, for example `x11-libs/libXft` comes before `x11-libs/libXpm`, `x11-apps/` before `x11-libs/` and so on.

**Review by ceamac — CHANGES_REQUESTED** [review]
Only a few minor issues remain.

The commit message should start with `x11-misc/slim: `, the prefix got lost while squashing.  Also you lost all the `Closes:` tags, please add them back.

Thank you!

File: x11-misc/slim/slim-1.3.9.ebuild
Comment (RobPearce) [reply x11-misc/slim/slim-1.3.9.ebuild:87]: Oh, OK. Once again I copied this from an existing ebuild after searching for examples of the readme.gentoo-r1 eclass :(

File: x11-misc/slim/slim-1.3.9.ebuild
Comment (RobPearce) [reply x11-misc/slim/slim-1.3.9.ebuild]: OK, though it raises a separate question, which is a matter of preference, I think:
The pam USE flag currently controls whether slimlock is built. This follows the logic in the upstream CMakeLists from when Nobuhiro-san added it (at V1.3.6). That was, I think, an error - the slimlock code needs PAM and so should not be built _without_ USE_PAM, but it's not required to build it because of PAM. This logic has changed upstream from V1.4.0 but I've retained the old behaviour through the ebuild. It might be considered better to introduce a different (package-specific) USE flag to control it, then encode that this USE flag needs USE=pam first. Of course, that would also be a change of behaviour, so might need a message, too... and I'm not sure it's really necessary.

Comment (ceamac) [reply x11-misc/slim/slim-1.3.9.ebuild]: Do you see any value in building with `pam` but without slimlock?  Does slimlock require any extra dependencies, or does it take a long time to build?  If not, no need to add an extra USE flag, tying slimlock to the same knob as pam sounds reasonable.

Comment (RobPearce) [reply x11-misc/slim/slim-1.3.9.ebuild]: I don't think there's any real benefit other than "removal of cruft" (since I suspect very few people actually use slimlock). It adds only one dependency (libXext) but from the looks of things that's actually already an indirect dependency (via libXft?) and the build step for slimlock is pretty quick (one C++ file and a link).
So yes, the current approach feels pragmatically sensible.

**Comment by ceamac** [comment]
`MissingSignOff: commit be9ed1856466, missing sign-off: Robert Pearce <gentoo@flitspace.org.uk>`

It appears pkgcheck is unhappy your name in the GCO is different from the one in Author.

Also the `~riscv` keyword is missing, it was added to the tree on Feb 28 and your PR is a bit older than that.  I can add it while merging, but I wouldn't want to touch your Signed-Off-By, so can you fix that please?

**Comment by RobPearce** [comment]
> Also the `~riscv` keyword is missing, it was added to the tree on Feb 28 and your PR is a bit older than that. I can add it while merging, but I wouldn't want to touch your Signed-Off-By, so can you fix that please?

It looks like pkgcheck doesn't like `~riscv` - should I revert that?

**Comment by ceamac** [comment]
No, it's ok.  Your branch is older than the commit where `x11-themes/slim-themes` was keyworded ~riscv.  If you rebase against master, the error goes away.

But the name in the Signed-off-by line is still different from the one in Author.

**Comment by RobPearce** [comment]
> But the name in the Signed-off-by line is still different from the one in Author.

Really? I was sure I'd changed it. Try now...

**Review by ceamac — APPROVED** [review]
Thank you!

**Comment by ceamac** [comment]
While trying to assign you the remaining bugs in bugzilla, it appears your bugzilla email is different from the one in Author and GCO.  Please change it to be the same, so the automated tools work properly.

`gentoo@flitspace.org.uk` vs `bugs.gentoo@flitspace.org.uk`

**Comment by RobPearce** [comment]
> While trying to assign you the remaining bugs in bugzilla, it appears your bugzilla email is different from the one in Author and GCO. Please change it to be the same, so the automated tools work properly.

OK, will do. I see you found some old bugs I hadn't seen, so I'll look see if there's anything to be done. Although the one raised against a 2017 vintage -9999 ebuild  may be tricky to follow up.

**Comment by RobPearce** [comment]
> Sorry, I meant the email in `metadata.xml`

I thought I'd set that to match the bugzilla one. Having "bugs" in the commit author email wouldn't make sense but the docs said the metadata one needs to match the bugzilla login.

**Comment by ceamac** [comment]
Then we're good.  Thank you!

**Pull request closed without merging**
